agent/containermetadata/types.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+const (
+	// MetadataInitial is the initial state of the metadata file which
+	// contains metadata provided by the ECS Agent
+	MetadataInitialText = "INITIAL"
+	// MetadataReady is the final state of the metadata file which indicates
+	// it has acquired all the data it needs (Currently from the Agent and Docker)
+	MetadataReadyText = "READY"
+)
+
+const (
+	MetadataInitial MetadataStatus = iota
+	MetadataReady
+)
+
+// MetadataStatus specifies the current update status of the metadata file.
+// The purpose of this status is for users to check if the metadata file has
+// reached the stage they need before they read the rest of the file to avoid
+// race conditions (Since the final stage will need to be after the container
+// starts up
+// In the future the metadata may require multiple stages of update and these
+// statuses should amended/appended accordingly.
+type MetadataStatus int32
+
+func (status MetadataStatus) MarshalText() (text []byte, err error) {
+	switch status {
+	case MetadataInitial:
+		text = []byte(MetadataInitialText)
+	case MetadataReady:
+		text = []byte(MetadataReadyText)
+	default:
+		return nil, fmt.Errorf("failed marshalling MetadataStatus %v", status)
+	}
+	return
+}
+
+func (status *MetadataStatus) UnmarshalText(text []byte) error {
+	var err error
```

Comment:
You can get rid of this line and `return nil` at the end of the method.

Reply:
oh i see why, fixed.